### PR TITLE
replace internet route-able ipv4s in docker-compose

### PR DESCRIPTION
**Pull request by gregnuj**
replace internet routeable ipv4s in docker-compose
ref: https://datatracker.ietf.org/doc/html/rfc1918
https://ip-netblocks.whoisxmlapi.com/lookup-report/8L2ojwyPRE

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by thanethomson — CHANGES_REQUESTED** [review]
Thanks for bringing this up @gregnuj - could you please explain the problem you're encountering to which this is the solution?

If this is a problem, there are two other changes that still need to be made for us to accept this:
1. The `localnet` target in the `Makefile` needs to be updated: https://github.com/cometbft/cometbft/blob/c3acdb866f885e94dcd0e1ef8ecce53ff689216c/Makefile#L287-L289
2. The [docs](https://github.com/cometbft/cometbft/blob/main/docs/networks/docker-compose.md) need to be updated

**Comment by sergio-mena** [comment]
Further on @thanethomson's answer, what is the problem we are trying to solve?
RFC1918 defines a set of network prefixes (private networks) that MUST NOT be routable outside their subnet (10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16), but it does not mandate anything about the rest of prefixes (you still still use any other prefix for your private network, and it's up to you if you want it to be reachable).
In short, (as mentioned at the beginning) I don't see the problem this PR is trying to solve.
